### PR TITLE
9658 - Replace Misc Default Type

### PR DESCRIPTION
**Pull request by codyseibert**

### CODE REVIEW

**Review by mrinsin — APPROVED** [review]

File: shared/src/business/entities/DocketEntry.test.js [hunk 1]
```diff
@@ -439,4 +439,69 @@ describe('DocketEntry entity', () => {
...
+          isDraft: true,
+          judgeUserId: mockJudgeUserId,
+        },
+        { applicationContext, petitioners: MOCK_PETITIONERS },
+      );
+
+      expect(docketEntry.isValid()).toBeTruthy();
+    });
+
+    it('eventCode should be optional if the document isDraft is true', () => {
```
Comment (rachelschneiderman) [inline shared/src/business/entities/DocketEntry.test.js]: Love this test! I think it's duplicated though?

File: web-client/src/presenter/actions/UploadCourtIssuedDocument/setDocumentTitleFromFreeTextAction.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import { state } from 'cerebral';
```
Comment (rachelschneiderman) [inline web-client/src/presenter/actions/UploadCourtIssuedDocument/setDocumentTitleFromFreeTextAction.js:1]: Tests?

File: shared/src/business/entities/EntityValidationConstants.js [hunk 3]
```diff
@@ -79,17 +79,32 @@ const DOCKET_ENTRY_VALIDATION_RULE_KEYS = {
   documentTitle: JoiValidationConstants.DOCUMENT_TITLE.optional().description(
     'The title of this document.',
   ),
-  documentType: JoiValidationConstants.STRING.valid(...ALL_DOCUMENT_TYPES)
-    .required()
-    .description('The type of this document.'),
+  documentType: joi.when('isDraft', {
```
Comment (rachelschneiderman) [inline shared/src/business/entities/EntityValidationConstants.js:82]: Tests?

File: shared/src/business/entities/DocketEntry.test.js [hunk 4]
```diff
@@ -439,4 +439,69 @@ describe('DocketEntry entity', () => {
...
+        {
+          ...A_VALID_DOCKET_ENTRY,
+          documentType: undefined,
+          eventCode: undefined,
+          isDraft: true,
+          judgeUserId: mockJudgeUserId,
+        },
+        { applicationContext, petitioners: MOCK_PETITIONERS },
+      );
+      docketEntry.validate();
```
Comment (rachelschneiderman) [inline shared/src/business/entities/DocketEntry.test.js]: Out of curiosity, why call validate first?

Comment (codyseibert) [reply shared/src/business/entities/DocketEntry.test.js]: that was actually a debug thing left over, good catch

File: web-client/src/presenter/actions/UploadCourtIssuedDocument/setDocumentTitleFromFreeTextAction.js (hunk 2 above)
Comment (codyseibert) [reply web-client/src/presenter/actions/UploadCourtIssuedDocument/setDocumentTitleFromFreeTextAction.js:1]: added one, ty

File: shared/src/business/entities/EntityValidationConstants.js (hunk 3 above)
Comment (codyseibert) [reply shared/src/business/entities/EntityValidationConstants.js:82]: added coverage

**Pull request merged**